flows/knack/knack_banner.py

```diff
@@ -4,124 +4,58 @@
 
 
 # Retrieve the provider's data
 @task(
-    name="HR knack banner integration",
-    max_retries=2,
-    retry_delay=timedelta(minutes=5),
-    state_handlers=[handler],
-    slug="knack-banner",
+    name="update HR employees in knack",
+    retries=2,
+    retry_delay_seconds=300,
+    task_run_name="atd-knack-banner/update-employees",
 )
-def knack_banner_update_employees():
+def knack_banner_update_employees(environment_variables):
+    logger = get_run_logger()
     response = (
         docker.from_env()
         .containers.run(
             image=docker_image,
             working_dir="/app",
             command=f"./atd-knack-banner/update_employees.py",
-            environment=environment_variables,
+            environment=environment_variables.value,
             volumes=None,
             remove=True,
             detach=False,
             stdout=True,
         )
         .decode("utf-8")
     )
-    logger = prefect.context.get("logger")
     logger.info(response)
     return response
 
 
-# Configure email task
-email_task = EmailTask(
-    name="email_task",
-    subject="Knack HR updates from Banner",
-    email_to=environment_variables["EMAIL_ADDRESS"],
-    email_to_cc=environment_variables["EMAIL_ADDRESS_CC"],
-    email_from=email_config["email_from"],
-    smtp_server=email_config["smtp_server"],
-    smtp_port=email_config["smtp_port"],
-    smtp_type=email_config["smtp_type"],
-    attachments=None,
-)
-
-
-@task(log_stdout=True, state_handlers=[handler])
-def format_email_body(flow_data):
-    flow_data_list = flow_data.split("\n")
-    info_list = []
-    # only email the logs that are labeled "INFO:root"
-    for line in flow_data_list:
-        if line[0:9] == "INFO:root":
-            info_list.append(line[10:] + "<br>")
-    return " ".join(info_list)
-
-
-with Flow(
-    f"atd_knack_banner_{current_environment}",
-    storage=GitHub(
-        repo="cityofaustin/atd-prefect",
-        path="flows/knack/knack_banner.py",
-        ref="main"
-    ),
-    run_config=UniversalRun(labels=[current_environment, "atd-data02"]),
-    result=PrefectResult(),
-    schedule=Schedule(clocks=[CronClock("45 13 * * *")])
-) as get_data_flow:
-    email_data = knack_banner_update_employees()
-
-
-with Flow(
-    f"send_hr_email_{current_environment}",
-    storage=GitHub(
-        repo="cityofaustin/atd-prefect",
-        path="flows/knack/knack_banner.py",
-        ref="main"  # The branch name
-    ),
-    run_config=UniversalRun(labels=[current_environment, "atd-data02"]),
-) as send_email_flow:
-    get_data_flow_run_id = create_flow_run(flow_name=get_data_flow.name)
-    script_result = get_task_run_result(
-        get_data_flow_run_id, task_slug="knack-banner-copy"
-    )
-    formatted_data = format_email_body(script_result)
-    send_email_flow.chain(script_result, formatted_data, email_task(msg=formatted_data))
+@flow(name=f"Knack HR Banner")
+def knack_hr_banner_flow():
+    logger = get_run_logger()
```

Comment:
Doesn't look like you are using this logger. I was sad to learning in my testing that Prefect 2 doesn't allow us to define loggers as a global so you either have to pass this logger around or get a new one in each task like what you have

Reply:
you're right, i didnt understand why the global logger wasnt working and this was left over from when i thought i was going to pass it instead of creating a new one in that task